### DISCUSSION TITLE
Playtime: Add new mode to continue clip where it left off

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#1038] [Ideas]
Check if this is orthogonal to trigger mode or not. Also think about who this relates to legato and gate.

**Comment by Acousmachin** [comment]
I don't understand what "orthogonal to trigger mode" means :-(

**Comment by helgoboss** [comment]
This was just a note to myself. Whenever I add a new feature / configuration option, I need to decide whether to (a) just make it a new option within an existing configuration property or (b) add a completely new configuration property. This decision has lots of consequences for the usability of Playtime. If this is not done correctly, Playtime will at some point in the future feel overloaded with many small disconnected features.

## Example

Here's my thinking process.

In this case, we already have the global "Trigger mode" configuration property in the matrix playback settings, which could potentially be a nice place to add the option "Continue". At the moment, it contains the options "Toggle", "Momentary" and "Retrigger". But if I add the option there, I should actually add 2 new options: "Continue Toggle" and "Continue Momentary" - both would make sense. At this point, I should maybe consider creating new configuration property, let's say "Continuation behavior" with possible options ("Continue" and "Restart"). This configuration option would be **orthogonal** to "Trigger mode", which means it's actually a different thing and can be applied in addition. The user can then combine as he wants.

But then, since the combination "Retrigger" + "Continue" wouldn't make much sense, this is probably also not the best possibility. So maybe I will just go with the following trigger modes:

- Toggle Play/Stop
- Momentary Play/Stop
- Toggle Play/Pause
- Momentary Play/Pause
- Retrigger

Looks quite consistent and easy to understand for users. But now, let's assume I want to add more similar options in the future. I can think of "Gate" (triggering the clip in a gate-like way, synchronized to the timeline). Then this list would become:

- Toggle Play/Stop
- Momentary Play/Stop
- Toggle Play/Pause
- Momentary Play/Pause
- Toggle Gate
- Momentary Gate
- Retrigger

We see there's a sort of explosion of combinations. But I still consider this as better than adding one additional configuration property. Because "Toggle" and "Momentary" don't make sense for "Retrigger".

**Feedback or other ideas welcome! It's much better to get such things right from the beginning.**

**Comment by Acousmachin** [comment]
Thanks for these precious explainations, now I understand ;-)
For me, adding Pay/Pause new trigger modes is what seems the more logical and easy to use.

**Comment by helgoboss** [comment]
Nice overview in duplicate discussion #1125.

**Reply by endlesssoperator-cpu** [reply]
dear Helgoboss, is legato/gated playing mode something you might consider adding to playtime 2? it is something essential to my workflow/style of composing and I would love to be able to use it in playtime and reaper! kind regards @helgoboss [btw I am the same guy as in discussion #1125 where I go by prtrjan]

**Reply by endlesssoperator-cpu** [reply]
in logic Live Loops the settings for the wanted behavior are: 
<img width="250" height="63" alt="Screenshot 2026-07-04 at 10 38 56" src="https://github.com/user-attachments/assets/96eddee2-355a-4d95-bfea-8a2cf8153994" />
with quantization off
